R/paginate.R

```diff
@@ -142,15 +157,43 @@ paginate_next_request <- function(resp, req) {
   next_request(resp = resp, req = req)
 }
 
+paginate_parse_body <- function(resp, req) {
+  f_body <- req$policies$paginate$body
+  if (!is.null(f_body)) {
+    resp$parsed_body <- f_body(resp)
+  }
+
+  resp
+}
+
+paginate_register_body <- function(f) {
+  out <- function(...) {
+    data <- list(body = resp$parsed_body)
+    f_env <- fn_env(f)
+    fn_env(f) <- new_environment(data = data, parent = f_env)
+
+    call <- rlang::call_match()
+    call[[1]] <- f
+    rlang::eval_bare(call)
+  }
+  rlang::fn_fmls(out) <- rlang::fn_fmls(f)
+
+  out
+}
+
 #' @param next_url A function that extracts the url to the next page from the
 #'   [response].
 #' @rdname req_paginate
 #' @export
 req_paginate_next_url <- function(req,
                                   next_url,
+                                  body = NULL,
                                   n_pages = NULL) {
   check_function2(next_url, args = "resp")
 
+  next_url_user <- next_url
+  next_url <- paginate_register_body(next_url_user)
```

Comment:
I'm not quite sure the approach in `paginate_register_body()` makes sense as I have to modify the `next_url` function instead of only modifying the `next_request` function in `req_paginate()`